azblob/sas_service.go

```diff
@@ -24,6 +25,83 @@ type BlobSASSignatureValues struct {
 	ContentEncoding    string // rsce
 	ContentLanguage    string // rscl
 	ContentType        string // rsct
+	UserDelegationKey  UserDelegationKey
+}
+
+func (v BlobSASSignatureValues) NewIdentitySASQueryParameters(AccountName string) (SASQueryParameters, error) {
+	if v.UserDelegationKey.Value == "" {
+		return SASQueryParameters{}, errors.New("please provide a UserDelegationKey")
+	}
+
+	resource := "b"
+	//Make sure the permission characters are in the correct order
+	perms := &BlobSASPermissions{}
+	if err := perms.Parse(v.Permissions); err != nil {
+		return SASQueryParameters{}, err
+	}
+	v.Permissions = perms.String()
+
+	if v.Version == "" {
+		v.Version = SASVersion
+	}
+	startTime, expiryTime := FormatTimesForSASSigning(v.StartTime, v.ExpiryTime)
+
+	udk := v.UserDelegationKey
+	udkStart, udkExpiry := FormatTimesForSASSigning(udk.SignedStart, udk.SignedExpiry)
+	udkString := strings.Join([]string{
+		udk.SignedOid,
+		udk.SignedTid,
+		udkStart,
+		udkExpiry,
+		udk.SignedService,
+		udk.SignedVersion,
+	}, "\n")
+
+	// String to sign: http://msdn.microsoft.com/en-us/library/azure/dn140255.aspx
+	stringToSign := strings.Join([]string{
```

Comment:
This method feels very redundant, please consider doing some refactoring so that there's less duplicated code between this method and NewSASQueryParameters.

Reply:
Example [here](https://github.com/Azure/azure-storage-python/blob/master/azure-storage-blob/azure/storage/blob/sharedaccesssignature.py#L232).

Reply:
Again, pulling in the previous ways I'd implemented this, I disliked that I was using signedIdentifier for two things. Again, I will revert this.